internal/controllers/routerinterface/reconcile.go

```diff
@@ -88,12 +87,17 @@ func (r *orcRouterInterfaceReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	routerInterfacePortIterator := networkClient.ListPort(ctx, &listOpts)
 	// We're going to iterate over all interfaces multiple times, so pull them all in to a slice
-	var routerInterfacePorts []ports.Port //nolint:prealloc // We don't know how many ports there are
+	var routerInterfacePorts []ports.Port
 	for port, err := range routerInterfacePortIterator {
 		if err != nil {
 			return ctrl.Result{}, fmt.Errorf("fetching router interface ports: %w", err)
 		}
-		routerInterfacePorts = append(routerInterfacePorts, *port)
+		switch port.DeviceOwner {
+		case "network:router_interface", "network:ha_router_replicated_interface", "network:router_interface_distributed":
```

Comment:
I wonder if we're setting ourselves up for now, similar failures. How would you feel about removing this entirely and just relying on the router uuid to be universally unique?

Reply:
I tried to be conservative with my patch as I don't know for sure if we're not going to see unrelated ports with the DeviceID set to the router UUID on the target subnet.

For reference, I built that list from https://github.com/openstack/neutron-lib/blob/950f0f3a8a722da115c03f2abb756a732a3ec4c1/neutron_lib/constants.py#L78-L80.